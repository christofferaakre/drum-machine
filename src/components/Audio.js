/* eslint jsx-a11y/media-has-caption: "off" */

import React from 'react';
import PropTypes from 'prop-types';

export const styles = {
  container: {
    padding: 15,
  },
};

export class Audio extends React.Component {
  componentDidMount() {
    const { keyToPress, } = this.props;

    this.audio = document.getElementById(keyToPress);
  }
  render() {
    const { src, keyToPress, volume, } = this.props;

    if (this.audio) {
      this.audio.volume = volume;
    }
    return (
      <div
        onClick={() => this.audio.play()}
        className='drum-pad'
        id={`drum-pad-container-${ src }`}
        style={styles.container}
      >
        {keyToPress || 'Click'}
        <audio volume={volume} className='clip' id={keyToPress} src={src} />
      </div>
    );
  }
}

Audio.propTypes = {
  src       : PropTypes.string.isRequired,
  keyToPress: PropTypes.string,
  volume    : PropTypes.number,
};

Audio.defaultProps = {
  keyToPress: '',
  volume    : 0.5,
};

export default Audio;
