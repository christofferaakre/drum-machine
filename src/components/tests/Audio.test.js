import React from 'react';
import { shallow, } from 'enzyme';
import { Audio, } from '../Audio';

let props;

beforeEach(() => {
  props = {
    src       : 'src',
    keyToPress: 'K',
  };
});

describe('Audio', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<Audio {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
