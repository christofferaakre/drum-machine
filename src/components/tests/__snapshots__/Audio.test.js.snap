// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio renders correctly 1`] = `
<article>
  <audio
    id="src"
    src="src"
  >
    Your browser does not support HTML5 audio
  </audio>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</article>
`;
